mlir/lib/Dialect/MIOpenOps/CppOutput/gridwise_convolution_implicit_gemm_v4r4.cpp

```diff
@@ -702,91 +700,187 @@ void ObtainModuleInfo(ModuleOp &m,
-            filterLayoutAttr.getValue()[i].dyn_cast<StringAttr>().getValue();
-        if (filterDim.str() == "k") {
-          dimKF = i;
-        }
-        auto outputDim =
-            outputLayoutAttr.getValue()[i].dyn_cast<StringAttr>().getValue();
-        if (outputDim.str() == "ko") {
-          dimKO = i;
-        }
-      }
+void obtainInput1VecGemmKVectorizable(
+    miopen::ConvOpType opType,
+    llvm::StringMap<std::pair<size_t, int64_t>> &dimIndexVal,
+    bool &input1GemmKVectorizable) {
+  // Vectorizable flag is opposite between forwad and bwd_data
+  if (opType == miopen::ConvOpType::Conv2DOpType) {
+    // When K is not the fastest changing dimension,
+    // gemmK dimension is vectorizable, gemmM is not, and vice versa.
+    // Vectorization width depending on which among C, Y, X be the fastest
+    // changing dimension.
+    if (dimIndexVal["k"].first == 3) {
+      input1GemmKVectorizable = false;
+    } else {
+      input1GemmKVectorizable = true;
+    }
+  } else if (opType == miopen::ConvOpType::Conv2DBwdDataOpType) {
+    // When K is the fastest changing dimension(3),
+    // gemmK dimension is vectorizable, gemmM is not, and vice versa.
+    // Vectorization width depending on length of K.
+    if (dimIndexVal["k"].first == 3) {
+      input1GemmKVectorizable = true;
+    } else {
+      input1GemmKVectorizable = false;
+    }
+  }
+}
 
-      // For filter tensor:
-      // When K is the fastest changing dimension(3),
-      // gemmK dimension is vectorizable, gemmM is not, and vice versa.
-      // Vectorization width depending on length of K.
-      if (dimKF == 3) {
-        input1GemmKVectorizable = true;
-      } else {
-        input1GemmKVectorizable = false;
-      }
+void obtainInput1VecLen(
+    miopen::ConvOpType opType,
+    llvm::StringMap<std::pair<size_t, int64_t>> &dimIndexVal, int64_t &vecLen) {
+  // Vectorization length logic is the same for forward and bwd_data
+  if (dimIndexVal["k"].first == 3) {
+    vecLen = dimIndexVal["k"].second;
+  } else if (dimIndexVal["k"].first == 0) {
+    // dimKF is the lowest changing dimension, which means dimC/dimY/dimX
+    vecLen = dimIndexVal["c"].second * dimIndexVal["y"].second *
+             dimIndexVal["x"].second;
+  } else if (dimIndexVal["k"].first == 1) {
+    // K's position is at 1, vectorization legnth is last two dimension
+    if (dimIndexVal["c"].first == 0) {
+      vecLen = dimIndexVal["y"].second * dimIndexVal["x"].second;
+    } else if (dimIndexVal["y"].first == 0) {
+      vecLen = dimIndexVal["c"].second * dimIndexVal["x"].second;
+    } else {
+      vecLen = dimIndexVal["c"].second * dimIndexVal["y"].second;
+    }
+  } else {
+    // K's position is 2, vectorization legnth is last dimension
+    if (dimIndexVal["c"].first == 3) {
+      vecLen = dimIndexVal["c"].second;
+    } else if (dimIndexVal["y"].first == 3) {
+      vecLen = dimIndexVal["y"].second;
+    } else {
+      vecLen = dimIndexVal["x"].second;
+    }
+  }
+}
```

Comment:
@asroy Could you do a sanity check for the filter `kcyx` layout vectorization length?

FYI `dimIndexVal` is the a String to pair mapping which holds the 
 - key: string representation of dimension, for example, k
 - value pair: first: index of the  dimension, for example, 0 for k in kcyx. 
                       second: value of the dimension, for example, 128 for number of filters.